File: BowlingKata/BowlingKata-Fsharp/Bowling.SpecFlow/StepDefinitions/BowlingSteps.fs
module Bowling.SpecFlow.StepDefinitions

open Bowling.SpecFlow.Drivers
open TechTalk.SpecFlow

[<Binding>]
type public BowlingSteps(driver : BowlingDriver) =
    let _driver : BowlingDriver = driver
    let [<Given>]``a new bowling game``() =
       _driver.NewGame()

    let [<When>]``all of my balls are landing in the gutter``() =
        _driver.Throw(0, 20)

    let [<When>]``all of my rolls are strikes``() =
        _driver.Throw(10, 12)
     
    let [<Then>]``my total score should be (\d+)`` score =
        _driver.CheckScore score

    let [<When>]``I roll (\d+)`` pins =
         _driver.Throw(pins, 1)
     
    let [<When>]``I roll (\d+) and (\d+)`` pins1 pins2 =
        _driver.Throw(pins1, pins2, 1)
     
    let [<When>]``I roll (\d+) times (\d+) and (\d+)`` count pins1 pins2 =
        _driver.Throw(pins1, pins2, count)
     
    let [<When>]``I roll the following series:(.*)`` (series:string) =
        _driver.ThrowSeries(series)
     
    let [<When>]``I roll`` (rolls:Table) =
        _driver.ThrowSeries(rolls)

File: BowlingKata/BowlingKata-MsTest/Bowling.SpecFlow/StepDefinitions/BowlingSteps.cs
using Bowling.SpecFlow.Drivers;
using TechTalk.SpecFlow;

namespace Bowling.SpecFlow.StepDefinitions
{
    [Binding]
    public class BowlingSteps
    {
        private readonly BowlingDriver _bowlingDriver;

        public BowlingSteps(BowlingDriver bowlingDriver)
        {
            _bowlingDriver = bowlingDriver;
        }

        [Given(@"a new bowling game")]
        public void GivenANewBowlingGame()
        {
            _bowlingDriver.NewGame();
        }

        [When(@"all of my balls are landing in the gutter")]
        public void WhenAllOfMyBallsAreLandingInTheGutter()
        {
            _bowlingDriver.Roll(0, 20);
        }

        [When(@"all of my rolls are strikes")]
        public void WhenAllOfMyRollsAreStrikes()
        {
            _bowlingDriver.Roll(10, 12);
        }

        [Then(@"my total score should be (\d+)")]
        public void ThenMyTotalScoreShouldBe(int score)
        {
            _bowlingDriver.CheckScore(score);
        }

        [When(@"I roll (\d+)")]
        public void WhenIRoll(int pins)
        {
            _bowlingDriver.Roll(pins, 1);
        }

        [When(@"I roll (\d+) and (\d+)")]
        public void WhenIRoll(int pins1, int pins2)
        {
            _bowlingDriver.Roll(pins1, pins2, 1);
        }

        [When(@"I roll (\d+) times (\d+) and (\d+)")]
        public void WhenIRollSeveralTimes2(int rollCount, int pins1, int pins2)
        {
            _bowlingDriver.Roll(pins1, pins2, rollCount);
        }

        [When(@"I roll the following series:(.*)")]
        public void WhenIRollTheFollowingSeries(string series)
        {
            _bowlingDriver.RollSeries(series);
        }

        [When(@"I roll")]
        public void WhenIRoll(Table rolls)
        {
            _bowlingDriver.RollSeries(rolls);
        }
    }
}
